packages/nep141-erc20/src/bridged-nep141/sendToEthereum/index.js

```diff
@@ -533,49 +549,46 @@ async function checkSync (transfer) {
     )
     return transfer
   }
-  const nearAccount = await getNearAccount()
 
   const nearOnEthClient = new web3.eth.Contract(
     JSON.parse(process.env.ethNearOnEthClientAbiText),
     process.env.ethClientAddress
   )
 
-  const finalityBlockHeight = last(transfer.finalityBlockHeights)
+  const withdrawBlockHeight = last(transfer.withdrawReceiptBlockHeights)
   const { currentHeight } = await nearOnEthClient.methods.bridgeState().call()
   const nearOnEthClientBlockHeight = Number(currentHeight)
 
-  if (nearOnEthClientBlockHeight <= finalityBlockHeight) {
+  if (nearOnEthClientBlockHeight <= withdrawBlockHeight) {
     return {
       ...transfer,
       nearOnEthClientBlockHeight,
       status: status.IN_PROGRESS
     }
   }
 
-  const clientBlockHashB58 = bs58.encode(toBuffer(
-    await nearOnEthClient.methods
-      .blockHashes(nearOnEthClientBlockHeight).call()
-  ))
-  const withdrawReceiptId = last(transfer.withdrawReceiptIds)
-  const proof = await nearAccount.connection.provider.sendJsonRpc(
-    'light_client_proof',
-    {
-      type: 'receipt',
-      receipt_id: withdrawReceiptId,
-      receiver_id: transfer.sender,
-      light_client_head: clientBlockHashB58
-    }
-  )
-
   return {
     ...transfer,
     completedStep: SYNC,
-    nearOnEthClientBlockHeights: [...transfer.nearOnEthClientBlockHeights, nearOnEthClientBlockHeight],
-    proofs: [...transfer.proofs, proof],
+    nearOnEthClientBlockHeight,
     status: status.ACTION_NEEDED
   }
 }
 
+/**
+ * Check if a NEAR outcome receipt_id has already been used to finalize a transfer to Ethereum.
+ * @param {*} web3
+ * @param {*} proof
+ */
+async function proofAlreadyUsed (web3, proof) {
+  const usedProofsKey = bs58.decode(proof.outcome_proof.outcome.receipt_ids[0]).toString('hex')
+  const usedProofsMappingPosition = '0'.repeat(63) + '3'
+  const storageIndex = web3.utils.sha3('0x' + usedProofsKey + usedProofsMappingPosition)
+  // eth_getStorageAt docs: https://eth.wiki/json-rpc/API
+  const proofIsUsed = await web3.eth.getStorageAt(process.env.ethLockerAddress, storageIndex)
+  return Number(proofIsUsed) === 1
+}
```

Comment:
Didi you tested manually that this is working?

Reply:
Yes, tested with multiple transfers